endpoints/openrtb2/auction_test.go

```diff
@@ -227,16 +229,74 @@ func TestBlacklistRequests(t *testing.T) {
 	tests.assert(t)
 }
 
+// TestRejectAccountRequired asserts we return a 400 code on a request that comes with no user id nor app id
+// if the `AccountRequired` field in the `config.Configuration` structure is set to true
+func TestRejectAccountRequired(t *testing.T) {
+	tests := map[string]*getResponseFromDirectory{
+		"8.1) Not required and not provided. Since not provided, not blacklisted": {
+			dir:           "sample-requests/account-required",
+			file:          "no-acct.json",
+			payloadGetter: getRequestPayload,
+			messageGetter: nilReturner,
+			expectedCode:  http.StatusOK,
+			accountReq:    false,
+		},
+		"8.2) Required, not provided and in consequence, not blacklisted": {
+			dir:           "sample-requests/account-required",
+			file:          "no-acct.json",
+			payloadGetter: getRequestPayload,
+			messageGetter: getMessage,
+			expectedCode:  http.StatusBadRequest,
+			accountReq:    true,
+		},
+		"8.3) Required, provided and not blacklisted": {
+			dir:           "sample-requests/account-required",
+			file:          "with-acct.json",
+			payloadGetter: getRequestPayload,
+			messageGetter: nilReturner,
+			expectedCode:  http.StatusOK,
+			aliased:       true,
+			accountReq:    true,
+		},
+		"8.4) Required, provided and blacklisted": {
+			dir:           "sample-requests/blacklisted",
+			file:          "blacklisted-acct.json",
+			payloadGetter: getRequestPayload,
+			messageGetter: getMessage,
+			expectedCode:  http.StatusBadRequest,
+			accountReq:    true,
+		},
+	}
+	for _, test := range tests {
+		test.assert(t)
+	}
+}
+
 // assertResponseFromDirectory makes sure that the payload from each file in dir gets the expected response status code
 // from the /openrtb2/auction endpoint.
 func (gr *getResponseFromDirectory) assert(t *testing.T) {
 	//t *testing.T, dir string, payloadGetter func(*testing.T, []byte) []byte, messageGetter func(*testing.T, []byte) []byte, expectedCode int, aliased bool) {
 	t.Helper()
-	for _, fileInfo := range fetchFiles(t, gr.dir) {
-		filename := gr.dir + "/" + fileInfo.Name()
-		fileData := readFile(t, filename)
+	var filename string
+	var fileData []byte
+	if gr.file == "" {
+		for _, fileInfo := range fetchFiles(t, gr.dir) {
+			filename = gr.dir + "/" + fileInfo.Name()
+			fileData = readFile(t, filename)
```

Comment:
I'd recommend extracting the code from L286-L293 into a different function rather than having redundant code or another approach would be to create a slice of all filenames to be processed and then have the for loop range over that.

Reply:
Done. Implemented the array that we discussed, rather than the function.